src/main/java/org/opensearch/ad/transport/ForwardADTaskTransportAction.java

```diff
@@ -125,6 +151,12 @@ protected void doExecute(Task task, ForwardADTaskRequest request, ActionListener
                     adTaskCacheManager.removeHistoricalTaskCache(detectorId);
                 } else {
                     adTaskCacheManager.removeRealtimeTaskCache(detectorId);
+                    // If hash ring changed like new node added when scale out, the realtime job coordinating node may
+                    // change, then we should clean up cache on old coordinating node.
+                    stateManager.removeBackpressureCounter(detectorId);
+                    featureManager.clear(detectorId);
+                    entityColdStarter.removeDoorKeeper(detectorId);
+                    priorityCache.get().removeDetectorCache(detectorId);
```

Comment:
Priority cache will be cleaned up in hourly cron. Ideally we can use similar way of single entity detector to clean up model cache when receive node added/removed event. Will create an issue to track this.

Reply:
https://github.com/opensearch-project/anomaly-detection/issues/210